scipy/special/_logsumexp.py

```diff
@@ -212,4 +212,65 @@ def softmax(x, axis=None):
     """
 
     # compute in log space for numerical stability
-    return np.exp(x - logsumexp(x, axis=axis, keepdims=True))
+    # subtraction can cause cancellation when x and logsumexp(x) are close to each other.
+    # return np.exp(x - logsumexp(x, axis=axis, keepdims=True))
+    return np.exp(log_softmax(x, axis=axis))
+
+
+def log_softmax(x, axis=None):
+    r"""
+    Logarithm of softmax function::
+
+        log_softmax(x) = log(softmax(x))
+
+    Parameters
+    ----------
+    x : array_like
+        Input array.
+    axis : int or tuple of ints, optional
+        Axis to compute values along. Default is None and softmax will be
+        computed over the entire array `x`.
+
+    Returns
+    -------
+    s : ndarray
+        An array the same shape as `x`. Exponential of the result will sum to 1 along the
+        specified axis.
+
+    Notes
+    -----
+    `log_softmax` is more accurate than ``np.log(softmax(x))`` with inputs that make 
+    `softmax` saturate (see examples below).
+    
+    .. versionadded:: 1.5.0
+
+    Examples
+    --------
+    >>> from scipy.special import log_softmax
```

Comment:
All examples should just do `import scipy.special as sc`.

Reply:
Thank you for your comments!
Using the same style in logsumexp, softmax, and log_softmax may be better. Do you think I should also rewrite examples in logsumexp and softmax?

Reply:
A follow-up PR would be welcome! Best to not modify those docstrings in this PR though.